lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,445 @@
+                                      [1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0]],
+                                     [[0.25, 0, 0.25, 0.25, 0.25],
+                                      [0.25, 0, 0.25, 0.25, 0.25],
+                                      [0, 0, 0.333333, 0.333333, 0.333333],
+                                      [0, 0, 0, 0.9, 0.1],
+                                      [0.333333, 0.333333, 0.333333, 0, 0]],
+                                     [[1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0],
+                                      [0, 0, 0, 1, 0.0],
+                                      [0, 0, 0, 0.333333, 0.666667]]])
+        weights_cube = self.weights_cube.copy(input_weights)
+        nbhooded_cube = self.weights_cube.copy()
+        plugin = CollapseMaskedNeighbourhoodCoordinate(
+            "projection_x_coordinate", weights_cube)
+        plugin.reweight_weights(nbhooded_cube, weights_cube)
+        self.assertArrayAlmostEqual(expected_weights, weights_cube.data)
+
+
+class Test_process(IrisTest):
+
+    """Test the process method of CollapseMaskedNeighbourhoodCoordinate"""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5)
+        self.cube = iris.util.squeeze(self.cube)
+        weights_data = np.array([[[0.8, 0.7, 0., 0, 0],
+                                  [0.7, 0.3, 0, 0, 0],
+                                  [0.3, 0.1, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0]],
+                                 [[0.2, 0.3, 1, 1, 1],
+                                  [0.3, 0.7, 1, 1, 1],
+                                  [0.7, 0.9, 1, 1, 1],
+                                  [1, 1, 1, 0.9, 0.5],
+                                  [1, 1, 1, 0.6, 0.2]],
+                                 [[0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0, 0],
+                                  [0, 0, 0, 0.1, 0.5],
+                                  [0, 0, 0, 0.4, 0.8]]])
+        topographic_zone_points = [50, 150, 250]
+        topographic_zone_bounds = [[0, 100], [100, 200], [200, 300]]
+
+        weights_cubes = iris.cube.CubeList([])
+        for data, point, bounds in zip(weights_data, topographic_zone_points,
+                                       topographic_zone_bounds):
+            weights_cubes.append(
+                set_up_topographic_zone_cube(
+                    data, point, bounds, num_grid_points=5))
+        self.weights_cube = weights_cubes.merge_cube()
+        self.plugin = CollapseMaskedNeighbourhoodCoordinate("topographic_zone",
+                                                            self.weights_cube)
+
```

Comment:
I think there should be a test_basic here to prove that the result of the plugin is an Iris Cube.

Reply:
Done